testsupport/space.go

```diff
@@ -0,0 +1,95 @@
+package testsupport
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/tiers"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/gofrs/uuid"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/stretchr/testify/require"
+)
+
+// NewSpace initializes a new Space object with the given value. If the targetCluster is nil, then the Space.Spec.TargetCluster is not set.
+func NewSpace(namespace, name, tierName string, targetCluster *wait.MemberAwaitility) *toolchainv1alpha1.Space {
+	space := &toolchainv1alpha1.Space{
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: namespace,
+			Name:      name,
+		},
+		Spec: toolchainv1alpha1.SpaceSpec{
+			TierName: tierName,
+		},
+	}
+	if targetCluster != nil {
+		space.Spec.TargetCluster = targetCluster.ClusterName
+	}
+	return space
+}
+
+// GenerateName appends generated UUID to the given string
+func GenerateName(prefix string) string {
+	return fmt.Sprintf("%s-%s", prefix, uuid.Must(uuid.NewV4()).String())
+}
+
+// VerifyResourcesProvisionedForSpaceWithTier verifies that the Space for the given name is provisioned with all needed labels and conditions.
+// It also checks the NSTemplateSet and all related namespace & cluster scoped resources
```

Comment:
This function is pretty close to `VerifyResourcesProvisionedForSpaceWithTiers`, it wasn't immediately obvious to me so maybe we should change this comment to
```suggestion
// Same as VerifyResourcesProvisionedForSpaceWithTiers but reuses the provided tier name for the namespace and cluster resources names
```